pkg/controller/perconaservermongodb/mgo.go

```diff
@@ -144,7 +189,23 @@ func (r *ReconcilePerconaServerMongoDB) reconcileCluster(cr *api.PerconaServerMo
 	return clusterInit, nil
 }
 
-func (r *ReconcilePerconaServerMongoDB) mongoClient(cr *api.PerconaServerMongoDB, replSet *api.ReplsetSpec, pods corev1.PodList, username, password string) (*mgo.Client, error) {
+func inShard(client *mgo.Client, podName string) (bool, error) {
+	shardList, err := mongo.ListShard(context.TODO(), client)
+	if err != nil {
+		return false, errors.Wrap(err, "unable to get shard list")
+	}
+
+	for _, shard := range shardList.Shards {
+		if strings.Contains(shard.Host, podName) {
```

Comment:
Should we also compare a replset name here? As I see it is still possible for the user to rename the replset so rs name could change but pod name remains the same.

Reply:
pod name contains rs name